src/fiatconnect/saga.ts

```diff
@@ -277,14 +280,57 @@ export function* handleSelectFiatConnectQuote({
   payload: params,
 }: ReturnType<typeof selectFiatConnectQuote>) {
   const { quote } = params
+
   try {
+    // If KYC is required for the quote, check that the user has correct KYC on file
+    // with the quote's provider
+    let getKycStatusResponse: GetKycStatusResponse
+    if (quote.requiresKyc()) {
+      const kycSchema = quote.getKycSchema()!
```

Comment:
nit: by using this assertion operator, we are baking in the assumption that if `quote.requiresKyc()` evaluates to true, this will be nonnull. Equivalently, we could make use of the logical contrapositive, e.g. if `quote.getKycSchema()` is falsy, kyc is not required. (And ditch the usage of `quote.requiresKyc()` plus the assertion operator type hack.) This seems preferable to me.

Reply:
maybe a rename like `getRequiredKycSchema` could make this suggestion more palatable (since it's fairly clear to me that if there isn't a required kyc schema, I don't have to do kyc)

Reply:
Yeah, I went back and forth between just having `getKycSchema` and using its falsiness as a proxy for knowing whether or not KYC is required, or using both `requiresKyc` and `getKycSchema`. I think using just `getKycSchema is fine; it will definitely make things cleaner.